pkg/spec/spec.go

```diff
@@ -119,7 +117,15 @@ type App struct {
 	// List of Kubernetes resource files, that can be directly given to Kubernetes
 	ExtraResources []string `json:"extraResources,omitempty"`
 
-	PodSpecMod                 `json:",inline"`
+	PodSpecMod `json:",inline"`
+}
+
+// DeploymentSpecMod is Kedge's extension of Kubernetes DeploymentSpec and allows
+// defining a complete kedge application
+type DeploymentSpecMod struct {
+	// Specify which type of Kubernetes controller to generate. Defaults to deployment.
+	Controller                 string `json:"controller,omitempty"`
```

Comment:
this should be part of `ControllerFields` struct right ? Since it is field that will be needed.

Reply:
@surajssd you added this in https://github.com/kedgeproject/kedge/commit/2e1bcf5b87a13dbd454a188d8d2397f3c91f2c37

This was required for the JSON schema stuff, right?

Reply:
Yes, but this should be in `ControllerFields` right ?

Reply:
Sure, I'm not aware how this works for the JSON schema bits. If it works in Controller fields, will put it there.
Moved.